### PR TITLE
Prevent cache issues on version updates

### LINKED ISSUE #58827: No docs for String on stable?
Hi! Looks like docs for std string methods are missing on stable release?

https://doc.rust-lang.org/std/string/struct.String.html#method.as_bytes

37906c37-5439-48fd-b536-8562abd30309

THe methods section is empty :( 

Everything works ok for beta.

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #58827.

cc @rust-lang/infra

### CODE REVIEW

**Comment by Mark-Simulacrum** [comment]
I presume you'll want to beta backport as well, but I'll leave that to the rustdoc and/or docs team to decide.

@bors r+

**Comment by QuietMisdreavus** [comment]
Beta-nominating since this fixes an issue we've had with docs version upgrades on the site for ages. @rust-lang/docs, would anyone like to approve the backport?

**Comment by ollie27** [comment]
@bors r-

This caused #59081 to fail because rustdoc-js is still looking for `main.js` but now it's named something like `main1.35.0.js`.

**Comment by GuillaumeGomez** [comment]
It needs to be merged after #58330.

**Comment by GuillaumeGomez** [comment]
#58330 has been merged, therefore this one should work now.

@bors: r=MarkSimulacrum

File: src/bootstrap/doc.rs [hunk 1]
```diff
@@ -343,12 +343,10 @@ fn invoke_rustdoc(
         .arg("--html-before-content").arg(&version_info)
         .arg("--html-in-header").arg(&favicon)
         .arg("--markdown-no-toc")
-        .arg("--markdown-playground-url")
-        .arg("https://play.rust-lang.org/")
-        .arg("-o").arg(&out)
-        .arg(&path)
-        .arg("--markdown-css")
-        .arg("../rust.css");
+        .arg("--resource-suffix").arg(crate::channel::CFG_RELEASE_NUM)
```
Comment (ollie27) [inline src/bootstrap/doc.rs]: `--resource-suffix` doesn't do anything for stand-alone markdown files.

File: src/bootstrap/doc.rs [hunk 2]
```diff
@@ -430,9 +428,9 @@ impl Step for Standalone {
                .arg("--html-before-content").arg(&version_info)
                .arg("--html-in-header").arg(&favicon)
                .arg("--markdown-no-toc")
+               .arg("--resource-suffix").arg(crate::channel::CFG_RELEASE_NUM)
```
Comment (ollie27) [inline src/bootstrap/doc.rs]: Again, `--resource-suffix` doesn't do anything for stand-alone markdown files.

**Review by ollie27 — CHANGES_REQUESTED** [review]
`--resource-suffix` needs to be passed to rustdoc when building the `test` and `proc_macro` crates as well.

**Comment by GuillaumeGomez** [comment]
Updated.

r? @ollie27

File: src/bootstrap/doc.rs [hunk 3]
```diff
@@ -589,6 +586,7 @@ impl Step for Test {
 
         cargo.arg("--no-deps")
              .arg("-p").arg("test")
+             .env("RUSTDOC_RESOURCE_SUFFIX", crate::channel::CFG_RELEASE_NUM)
```
Comment (ollie27) [inline src/bootstrap/doc.rs]: `src/bootstrap/bin/rustdoc.rs` also needs to read this environment variable.

**Comment by Dylan-DPC** [comment]
ping from triage @GuillaumeGomez any updates?

**Comment by GuillaumeGomez** [comment]
I need to apply @ollie27's comments.

**Comment by GuillaumeGomez** [comment]
And updated!

**Comment by Mark-Simulacrum** [comment]
Er, just saw that Travis is failing.

@bors r-

**Comment by GuillaumeGomez** [comment]
Yep, an issue with a theme file apparently? I'll check this evening.

**Comment by GuillaumeGomez** [comment]
Ok, it was the index generator generation which failed because the theme files didn't receive the suffix. Fixed now that we can move on!

@bors: r=Mark-Simulacrum,ollie27

**Comment by pietroalbini** [comment]
Ping @rust-lang/docs @rust-lang/rustdoc this needs to be approved for beta backport.

**Comment by GuillaumeGomez** [comment]
I approve for it. cc @QuietMisdreavus

**Comment by pietroalbini** [comment]
This was removed from the backport since it caused a build failure.

**Comment by ollie27** [comment]
This will need at least 37ab3dc5b3560792ae4eab0521e0c08bbbdd95d8 to be backported as well. That commit got attached to the wrong PR: #58330.

**Pull request merged**